src/cmd.cpp

```diff
@@ -140,6 +155,42 @@ static int subcommand_dem2tintiles(bool need_help,
 
     const std::string meshing_method = local_varmap["method"].as<std::string>();
 
+    if("zemlya" != meshing_method && "terra" != meshing_method)
+    {
+        throw po::error(std::string("unknown method ") + meshing_method);
+    }
+
+    char* emergency_memory = new char[16384];
```

Comment:
Could you please clarify this buffer allocation? Also you don't free it when no bad_alloc happens. In order to be consistent with the rest of codebase, please use unique_ptr instead of a raw pointer.

Reply:
The reason for this is, that if an OOM happens, the system might not have enough memory to actually handle and print the error message. Therefore 16k are allocated here, to be freed later (will add the free at the end of the code) to have enough memory to print the error.

Reply:
moved this to unique_ptr and added the same logic further below where I forgot to add it